static_src/components/route_list.jsx

```diff
@@ -2,92 +2,177 @@
 import RouteForm from './route_form.jsx';
 import RouteStore from '../stores/route_store.js';
+import SpaceStore from '../stores/space_store.js';
 
 import createStyler from '../util/create_styler';
-
-function stateSetter(appGuid) {
-  const routes = RouteStore.getAll();
-  const appRoutes = routes.filter((route) => route.app_guid === appGuid);
+import formatRoute from '../util/format_route';
+
+function stateSetter() {
+  const routes = RouteStore.getAll().map((route) => {
+    const domain = DomainStore.get(route.domain_guid);
+    if (!domain) return route;
+    return Object.assign({}, route, { domain_name: domain.name });
+  });
+  const appGuid = AppStore.currentAppGuid;
+  const appRoutes = routes.filter((route) => route.app_guid === appGuid)
+    .map((route) => {
+      if (route.path && (route.path[0] === '/')) {
+        route.path = route.path.replace('/', '');
+      }
+
+      return route;
+    });
 
   return {
-    routes: appRoutes
+    appGuid,
+    routes: appRoutes,
+    spaceGuid: SpaceStore.currentSpaceGuid,
+    showCreateForm: RouteStore.showCreateRouteForm
   };
 }
 
-window.RouteStore = RouteStore;
-
 export default class RouteList extends React.Component {
   constructor(props) {
     super(props);
     this.props = props;
-    this.state = stateSetter(props.initialAppGuid);
-    this._onChange = this._onChange.bind(this);
+    this.state = stateSetter();
     this.styler = createStyler(style);
+
+    this._onChange = this._onChange.bind(this);
+    this._createRoute = this._createRoute.bind(this);
+    this._createRouteAndAssociate = this._createRouteAndAssociate.bind(this);
+    this._addCreateRouteForm = this._addCreateRouteForm.bind(this);
+    this._removeCreateRouteForm = this._removeCreateRouteForm.bind(this);
   }
 
   componentDidMount() {
+    DomainStore.addChangeListener(this._onChange);
     RouteStore.addChangeListener(this._onChange);
   }
 
-  componentWillReceiveProps(nextProps) {
-    this.setState(stateSetter(nextProps.initialAppGuid));
+  componentWillReceiveProps() {
+    this.setState(stateSetter());
   }
 
   componentWillUnmount() {
+    DomainStore.removeChangeListener(this._onChange);
     RouteStore.removeChangeListener(this._onChange);
   }
 
   _onChange() {
-    this.setState(stateSetter(this.props.initialAppGuid));
+    this.setState(stateSetter());
   }
 
-  _handleRouteAction(routeGuid, ev) {
-    ev.preventDefault();
+  _toggleEditRoute(routeGuid, event) {
+    event.preventDefault();
     routeActions.toggleEdit(routeGuid);
   }
 
+  _addCreateRouteForm(event) {
+    if (event) event.preventDefault();
+    routeActions.showCreateForm();
+  }
+
+  _removeCreateRouteForm() {
+    routeActions.hideCreateForm();
+  }
+
+  _createRoute(route) {
+    const { spaceGuid } = this.state;
+    const domainGuid = route.domain_guid;
+    routeActions.createRoute(domainGuid, spaceGuid, route);
+    // routeActions.createRouteAndBindForApp(appGuid, domainGuid, spaceGuid);
```

Comment:
Yup, removed. Actually that whole function is no longer relevant so I pulled it out.